### DISCUSSION TITLE
Set 'time.octopus_energy_a_xxxxxxxa_intelligent_ready_time' with a fixed time

### DISCUSSION THREAD

**Discussion by garethcook** [BottlecapDave/HomeAssistant-OctopusEnergy#646] [Q&A]
I've been using a different Intelligent add-on for a while, no issues. I've decided to retire that and only use this add-on, but I'm running into an issue setting the target ready time. This add-on surfaces this element differently, and the goal here is just set the intelligent target ready time to 11am

I've currently set it to:
`- service: time.set_value
target:
entity_id: time.octopus_energy_a_0000000a_intelligent_ready_time
data:
time: “11:00:00”
alias: OI Target Time 11am`

Which produces an error " has an action that calls an unknown service: time.set_value." yet does run, but would appear to be malformed. I've looked at the octopus_energy.update_target_config option, but tbh I'm lost a little on how that could be used with this entity to set a fixed time. I've tried using generic text, etc, won't work.

If I choose 'Octopus Energy: Update target rate config' in the automation UI, the 'time.octopus_energy_a_0000000a_intelligent_ready_time' entity is not available for selection

is there a simple example? I've searched in this discussion group but nothing was obvious. Thanks!

**Comment by BottlecapDave** [comment]
You might be the first person to use this service in that way (most people use the UI to my knowledge). Do you have any errors in your logs when calling the service? You want an error containing `vehicleChargingPreferences`. Alternatively you want errors relating to `ready_time.py`.

When I do it through the UI in the developer tools, you'll want the structure to be (which I think you have)

```
service: time.set_value
target:
  entity_id: time.octopus_energy_a_0000000a_intelligent_ready_time
data:
  time: "11:00:00"
```

It _might_ be that the entity provides the value as a proper python type, whereas the service doesn't convert and just provide a string.

The update target rate service is for the target rate sensors.

**Comment by garethcook** [comment]
No errors in the logs with ready_time.py nor vehicleChargingPreferences

This is what comes up in the UI as a warning:
`OI Smart uses an unknown service
The automation "OI Smart" (automation.oi_check_everything) has an action that calls an unknown service: time.set_value.
This error prevents the automation from running correctly. Maybe this service is no longer available, or perhaps a typo caused it.
To fix this error, [edit the automation](http://homeassistant.local:8123/config/automation/edit/1675334409558) and remove the action that calls this service.
Click on SUBMIT below to confirm you have fixed this automation.`

In the logs:
`2023-12-19 08:32:00.503 WARNING (MainThread) [homeassistant.helpers.service] Referenced entities number.octopus_energy_a_xxxxxxxxxa_intelligent_ready_time are missing or not currently available
2023-12-19 08:32:00.693 ERROR (MainThread) [homeassistant.components.automation.oi_check_everything] OI Smart: Parallel action at step 1: parallel 2: Set Octopus Target Time To Charge to 11am: OI Target Time 11am Check: Error executing script. Service not found for call_service at pos 2: Service time.set_value not found.`

**Reply by BottlecapDave** [reply]
That is odd, as `time.set_value` is provided by HA and not this integration. I assume it doesn't autocomplete in the developer tools? I assume you can't set it by creating an automation in the UI?

**Reply by garethcook** [reply]
I can, via the UI, call the service 'Time: Set Time' and specify this target entity. It allows it, which is why I just went with it. This response [https://community.home-assistant.io/t/unknown-service-time-set-value/658098](url) someone said : "There is no native Home Assistant integration for time. The closest is input_datetime so this integration and it’s service calls must have been provided by a 3rd party integration."

On one hand it's valid for entry, on the other doesn't exist

**Reply by garethcook** [reply]
<img width="1163" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/29944592/188bbad0-157a-4bf6-80fa-98e68019b559">

**Reply by garethcook** [reply]
To clarify - the UI allows time.setvalue, it works, but it also complains

**Reply by BottlecapDave** [reply]
This is a service provided by HA as part of last year's version as seen in these [docs](https://www.home-assistant.io/integrations/time/#time-services). All services that are provided by this integration are with the octopus_energy domain. It feels like something has got corrupt in your HA instance or something hasn't registered property.

**Reply by BottlecapDave** [reply]
Could you provide the full automation to see anything jumps out?

**Reply by garethcook** [reply]
I can see time.set_value in the HA docs, which says 
"Building block integration
The time integration cannot be directly used. You cannot create your own time entities using this integration. This integration is a building block for other integrations to use, enabling them to create time entities for you."
https://www.home-assistant.io/integrations/time/

Tells me you can use it, but I can't?
Right now, smart charging is off but it's still reporting that I have an intelligent slot - I shouldn't. I'm going to try switching to the other plugin to see what results I get, and remove this one. My script was pretty robust before switching.

**Reply by BottlecapDave** [reply]
That means you can't build your own time entity like through a template, but the service should be available to you. I can call it on my instance, but I get an error from the OE APIs because I'm not on intelligent. I assume you're on a recent version of HA?

**Reply by garethcook** [reply]
I'm on 2023.12.3, everything up to date

**Reply by BottlecapDave** [reply]
I'm really stumped why it's not working for you. When you said "the UI allows time.setvalue, it works, but it also complains", what do you mean? It runs but throws an error or something else? Do you have any other errors or warnings in your logs that might relate to either the set_value service or this integration? Or maybe some other errors which might indicate why the service isn't being registered properly.

**Reply by garethcook** [reply]
In my first post:
...which produces an error " has an action that calls an unknown service: time.set_value." yet does run.

It works, but it complains in the logs, and prompts for a repair in the Settings UI page : "has an action that calls an unknown service: time.set_value."

**Reply by BottlecapDave** [reply]
Sorry I thought you were referring to calling the service via the developer tools. If you call it through that, what errors do you get?

**Reply by garethcook** [reply]
I go to Dev tools, Services, use the UI to select the Time:Set Time, works OK and I get a green tick.

**Reply by BottlecapDave** [reply]
hmm that is weird that you can call the service directly without issue, update the entity (I assume) without issue, but can't have an automation calling the same service. I've just set up a local automation with a timer trigger calling the service and I'm not seeing the error you are.

**Comment by gwilford** [comment]
As a data point only, I set intelligent ready time via HA's node-red extension using the call service node, however I don't specify seconds in the data payload, e.g. time: "11:00"
